internal/commands/macho/objc.go

```diff
@@ -1179,5 +1194,55 @@ func (o *ObjC) nonBuiltInType(typ string) string {
 		return o.nonBuiltInType(after)
 	}
 
-	return typ
+	if idx := strings.Index(typ, "*"); idx != -1 {
+		typ = typ[:idx]
+	}
+	return strings.TrimSpace(typ)
+}
+
+func (o *ObjC) contextualSplit(typ string, lDelim string, mDelim string, rDelim string) (string, []string, string) {
+	lDelimIdx := strings.Index(typ, lDelim)
+	rDelimIdx := strings.Index(typ, rDelim)
+
+	lDelimIt := lDelimIdx + strings.Index(typ[lDelimIdx+1:], lDelim) + 1
+	for lDelimIt < rDelimIdx {
+		rDelimIdx += strings.Index(typ[rDelimIdx+1:], rDelim) + 1
+
+		lDelimNext := strings.Index(typ[lDelimIt+1:], lDelim)
+		if lDelimNext == -1 {
+			break
+		}
+
+		lDelimIt += lDelimNext + 1
+	}
+	rDelimIdx += 1
+
+	l := typ[:lDelimIdx]
+	m := typ[lDelimIdx+1 : rDelimIdx-1]
```

Comment:
this line panics for certain frameworks

Reply:
panic: runtime error: slice bounds out of range [11:2]

Reply:
Another go-macho bug: https://github.com/blacktop/go-macho/pull/60

Reply:
think we should armor this check and throw and error incase of some regression in the future?

Reply:
@blacktop, just added some checks to contextualSplit